bpf/proxy/syncer.go

```diff
@@ -418,6 +430,12 @@ func (s *Syncer) applyDerived(sname k8sp.ServicePortName, t svcType, sinfo k8sp.
 		if err := s.writeSvc(sinfo, svc.id, count, local); err != nil {
 			return err
 		}
+		if svcTypeExternalIP == t {
+			err := s.writeLBSrcRangeSvcNATKey(sinfo, svc.id, count, local)
+			if err != nil {
+				log.Debugf("Failed to write LB source range NAT keys")
+			}
+		}
```

Comment:
You could promote this block into the caller (`apply()`) add a type `svcTypeExternalIPSrc` or something and create keys into the map that also include the src prefix...

Reply:
As we need to write keys with src ip range for every external IP, I feel the current approach would be better. Let me know your thoughts